packages/api/src/interfaces/endpoints/eventSub.external.ts

```diff
@@ -33,18 +41,67 @@ export interface HelixEventSubWebSocketTransportData {
 	method: 'websocket';
 
 	/**
-	 * The callback URL to send event notifications to.
+	 * The session ID of the websocket.
 	 */
 	session_id: string;
 
 	/**
 	 * The time when the client initiated the socket connection.
 	 */
 	connected_at: string;
+
+	/**
+	 * The time when the socket connection was lost.
+	 */
+	disconnected_at: string;
 }
 
 /** @private */
-export type HelixEventSubTransportData = HelixEventSubWebHookTransportData | HelixEventSubWebSocketTransportData;
+export interface HelixEventSubConduitTransportData {
+	/**
+	 * The type of transport.
+	 */
+	method: 'conduit';
+
+	/**
+	 * The conduit ID.
+	 */
+	conduit_id: string;
+}
+
+/** @private */
+export interface HelixEventSubConduitData {
+	/**
+	 * The conduit ID.
+	 */
+	id: string;
+	/**
+	 * The shard count.
+	 */
+	shard_count: number;
+}
+
+/** @private */
+export interface HelixEventSubConduitShardData {
+	/**
+	 * The shard ID.
+	 */
+	id: string;
+	/**
+	 * The shard status.
+	 */
+	status: HelixEventSubSubscriptionStatus;
+	/**
+	 * The transport method.
+	 */
+	transport: HelixEventSubTransportData;
+}
+
+/** @private */
+export type HelixEventSubTransportData =
+	| HelixEventSubWebHookTransportData
+	| HelixEventSubWebSocketTransportData
+	| HelixEventSubConduitTransportData;
```

Comment:
I don't think conduits can be nested into other conduits. If that's true, this should be reflected by this type (and the type name should also reflect its use here).

Reply:
Sorry, I'm not following here.
If I understand you correctly: Conduits are a third transport type now, along with webhook and websockets when calling /helix/eventsub/subscriptions. They made sense to add here.

Raw (and anonymized) response from Helix:

```
{
    "total": 1234,
    "data": [
    {
        "id": "random-guid",
        "status": "enabled",
        "type": "channel.chat.message",
        "version": "1",
        "condition": {
            "broadcaster_user_id": "1234",
            "user_id": "5678"
        },
        "created_at": "2024-03-08T02:29:13.67322332Z",
        "transport": {
            "method": "conduit",
            "conduit_id": "random-guid"
        },
        "cost": 0
    },
    {
        "id": "random-guid",
        "status": "enabled",
        "type": "channel.hype_train.progress",
        "version": "1",
        "condition": {
            "broadcaster_user_id": "1234"
        },
        "created_at": "2021-10-18T19:51:04.304359423Z",
        "transport": {
            "method": "webhook",
            "callback": "https://my.callback/url"
        },
        "cost": 0
    }
],
    "max_total_cost": 10000,
    "total_cost": 1234,
    "pagination": {}
}
```

Reply:
It just seems that the only place this type is used is in the conduit shards response, so having the conduit transport type here doesn't seem to make sense.

Also, your example uses private properties from the outside. Please never write production code like this, as there are no guarantees about these, ever.

Reply:
It's on [HelixEventSubSubscriptionData](https://github.com/twurple/twurple/blob/2b02bf274c70efb968952d0597cfd89437591032/packages/api/src/interfaces/endpoints/eventSub.external.ts#L66) as well. This is an existing type already, I just added conduit data to it.

I see what you mean now though about nesting conduits. I'll fix that.

And yeah, that was just an example. I don't have anything in prod using private properties.